src/main.rs

```diff
@@ -18,9 +26,105 @@ async fn main() {
         .await
         .expect("failed to get info");
 
-    // We only print it here, note that in real-life code you may want to call `.into_inner()` on
-    // the response to get the message.
-    println!("{info:#?}");
+    let pubkey = PublicKey::from_str(&info.into_inner().identity_pubkey).unwrap();
+    println!("Starting lndk for node: {pubkey}");
+
+    let _node_signer = LndNodeSigner::new(pubkey, client.signer());
+}
+
+struct LndNodeSigner<'a> {
+    pubkey: PublicKey,
+    secp_ctx: Secp256k1<secp256k1::All>,
+    signer: RefCell<&'a mut tonic_lnd::SignerClient>,
+}
+
+impl<'a> LndNodeSigner<'a> {
+    fn new(pubkey: PublicKey, signer: &'a mut tonic_lnd::SignerClient) -> Self {
+        LndNodeSigner {
+            pubkey,
+            secp_ctx: Secp256k1::new(),
+            signer: RefCell::new(signer),
+        }
+    }
+}
+
+impl<'a> NodeSigner for LndNodeSigner<'a> {
+    /// Get node id based on the provided [`Recipient`].
+    ///
+    /// This method must return the same value each time it is called with a given [`Recipient`]
+    /// parameter.
+    ///
+    /// Errors if the [`Recipient`] variant is not supported by the implementation.
+    fn get_node_id(&self, recipient: Recipient) -> Result<PublicKey, ()> {
+        match recipient {
+            Recipient::Node => Ok(self.pubkey),
+            Recipient::PhantomNode => Err(()),
+        }
+    }
+
+    /// Gets the ECDH shared secret of our node secret and `other_key`, multiplying by `tweak` if
+    /// one is provided. Note that this tweak can be applied to `other_key` instead of our node
+    /// secret, though this is less efficient.
+    ///
+    /// Errors if the [`Recipient`] variant is not supported by the implementation.
+    fn ecdh(
+        &self,
+        recipient: Recipient,
+        other_key: &PublicKey,
+        tweak: Option<&Scalar>,
+    ) -> Result<SharedSecret, ()> {
+        match recipient {
+            Recipient::Node => {}
+            Recipient::PhantomNode => return Err(()),
+        }
+
+        // Clone other_key so that we can tweak it (if a tweak is required). We choose to tweak the
+        // `other_key` because LND's API accept a tweak parameter (so we can't tweak our secret).
+        let tweaked_key = if let Some(tweak) = tweak {
+            other_key
```

Comment:
Clippy unhappy here: 
```
warning: using `clone` on type `bitcoin::secp256k1::PublicKey` which implements the `Copy` trait
  --> src/main.rs:84:13
   |
84 | /             other_key
85 | |                 .clone()
   | |________________________^ help: try dereferencing it: `(*other_key)`
   |
```
But dereferencing it gives me: `error[E0614]: type bitcoin::secp256k1::PublicKey cannot be dereferenced`

Reply:
You can drop the `.clone()` entirely and don't bother dereferencing here.

Reply:
> You can drop the .clone() entirely and don't bother dereferencing here.

Works! I think it was me who's guitly of being confused, not poor clippy :')